### PR TITLE
Remove C++03 zombies

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by hernando** [comment]
You have to rebase, I think the first commit was already merged.

File: lunchbox/mtQueue.ipp [hunk 1]
```diff
@@ -21,99 +21,81 @@ namespace lunchbox
 template <typename T, size_t S>
 MTQueue<T, S>& MTQueue<T, S>::operator=(const MTQueue<T, S>& from)
```
Comment (hernando) [inline lunchbox/mtQueue.ipp]: Add a move assignment and constructor?

File: lunchbox/mtQueue.ipp [hunk 2]
```diff
@@ -21,99 +21,81 @@ namespace lunchbox
...
-    _cond.lock();
-    while (_queue.size() > maxSize)
-        _cond.wait();
+    std::unique_lock<std::mutex> lock(_mutex);
+    _condition.wait(lock, [&] { return _queue.size() <= maxSize; });
+
     _maxSize = maxSize;
-    _cond.signal();
-    _cond.unlock();
+    _condition.notify_one();
```
Comment (hernando) [inline lunchbox/mtQueue.ipp]: why not notify_all?

File: lunchbox/monitor.h [hunk 3]
```diff
@@ -375,62 +359,44 @@ class Monitor
...
-        {
-            const T current = _value;
-            if (predicate(current))
-                return current;
-        }
-        ScopedCondition mutex(_cond);
-        while (!predicate(_value))
-            _cond.wait();
+    const T _wait(const F& predicate) const
+    {
```
Comment (eile) [inline lunchbox/monitor.h]: Premature optimization which would be harder to retain with C++11 primitives.

File: lunchbox/mtQueue.ipp (hunk 1 above)
Comment (eile) [inline lunchbox/mtQueue.ipp]: done

**Pull request merged**